### PR TITLE
OCPBUGS-27788: PowerVS: COS region configurable

### PR DESCRIPTION
**Pull request by hamzy**
In PowerVS, a region might not support COS objects.  Therefore, introduce a mapping.

Also, in making the Madrid region deployable, a number of bugs were found besides the big one of needing to use a different COS region.

### CODE REVIEW

**Comment by mjturek** [comment]
verify-codegen jobs are failing but that seems to be unrelated to change

File: pkg/asset/machines/worker.go [hunk 1]
```diff
@@ -179,13 +179,28 @@ func defaultVSphereMachinePoolPlatform() vspheretypes.MachinePool {
...
 		ProcType:   machinev1.PowerVSProcessorTypeShared,
 		SysType:    "s922",
 	}
+
+	sysTypes, err = powervstypes.AvailableSysTypes(ic.PowerVS.Region)
+	if err != nil {
+		defaultMp.SysType = sysTypes[0]
+	} else {
+		logrus.Warnf("For given region %v, AvailableSysTypes returns %v", ic.PowerVS.Region, err)
+	}
```
Comment (r4f4) [inline pkg/asset/machines/worker.go:201]: Isn't the logic here reversed?
```suggestion
	if err == nil {
		defaultMp.SysType = sysTypes[0]
	} else {
		logrus.Warnf("For given region %v, AvailableSysTypes returns %v", ic.PowerVS.Region, err)
	}
```

Comment (hamzy) [reply pkg/asset/machines/worker.go:201]: Weird!  I know I've tested this code in mad02 which should only have a non-922 system type.
Yes, the logic is reversed.

**Review by r4f4 — COMMENTED** [review]
/approve

**Pull request merged**